### PR TITLE
Feature:  New data elements added for multiple countries up to 40

### PR DESCRIPTION
**Pull request by davidbgomes**
Related to task: [Resources Comparative analysis: new data elements added for multiple countries up to 40](https://app.clickup.com/t/865cg9e8b)

Add new data elements to all the 40 related country, build the new comparative-analysis object, change filter logic of countries in order to filter by country OR related country/countries.

### CODE REVIEW

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @davidbgomes 

The GitHub action has failed executing yarn tsc, can you review it?

**Comment by davidbgomes** [comment]
Thanks @xurxodev , it's corrected now.

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @davidbgomes 

- I think we can improve the readability of the code in this PR. If you extract a new function getRelatedCountries the logic to extract countries from an event that currently is in buildComparativeAnalisis, then you can simplify filterCountries to a code similar to this:
```typescript
    private filterCountries(events: D2Event[], filters: ResourceFilters) {
        if (filters.category === "comparative-analysis") {

            
            return events.filter(event =>
                {
                    const relatedCountries = getRelatedCountries(event)

                    return relatedCountries?.some(country => filters.countries?.includes(country))
                }
            );
        } else {
            return events.filter(el =>
                el.dataValues.some(
                    el => el.dataElement === resourceDataElements.countryId && filters.countries?.includes(el.value)
                )
            );
        }
    }
```

- Can you add some tests in ResourceD2Repository.spec.ts to validate the new filter logic by countries

**Comment by davidbgomes** [comment]
Thank you @xurxodev ! Sounds good, I did the improvement of the filters readability (although I've kept the logic of the 'main' country and related countries in this case) & added a validation to the tests on `verifyComparativeAnalysis` to check for a couple related countries, that's what I could think of. We later could add more to `ResourcesByTypePage.spec` once the filters functionality testing is ok. Let me know if you think of something else to add/fix

**Comment by davidbgomes** [comment]
Thank you @xurxodev , makes sense I've joined the logic for the countries (first and all the other related), and added the tests for all related countries dataElements.

File: src/data/repositories/resources/ResourceD2Repository.ts [hunk 1]
```diff
@@ -140,9 +183,136 @@ export class ResourceD2Repository implements ResourceRepository {
     private buildComparativeAnalysis(event: D2Event): ComparativeAnalysis {
         const analysis = this.buildAnalysis(event);
 
+        const relatedCountry1 =
```
Comment (xurxodev) [inline src/data/repositories/resources/ResourceD2Repository.ts:186]: This code be moved to related countries and replaced by this:
const relatedCountries = this.getRelatedCountries();

You can review this branch where I have realized the change
feature/comparative-new-data-countries-jorge

**Review by xurxodev — CHANGES_REQUESTED** [review]
thanks @davidbgomes 

The code still has two places where retrieve related countries from an event.

I add comment in the code

File: src/data/repositories/resources/ResourceD2Repository.ts (hunk 1 above)
Comment (davidbgomes) [reply src/data/repositories/resources/ResourceD2Repository.ts:284]: Oh ok thank you, I see what you mean now.

**Comment by davidbgomes** [comment]
Thanks @xurxodev , I've cherry-picked the improvements you added on the other branch.

**Review by xurxodev — APPROVED** [review]
thanks @davidbgomes

**Pull request merged**